Library/Homebrew/formula.rb

```diff
@@ -1165,6 +1165,12 @@ def link_overwrite?(path)
   # @return [Boolean]
   delegate deprecated?: :"self.class"
 
+  # The date that this {Formula} was or becomes deprecated.
+  # Returns `nil` if no date is specified.
+  # @!method deprecation_date
+  # @return Date
+  delegate deprecation_date: :"self.class"
```

Comment:
Might be nice to have a `delegate ... :"self.class"` helper method because it's used in a tonne of places (as a follow-up PR).

Reply:
Agreed but I'd opt for the follow-up PR.

Reply:
Yup, thanks!